internal/pkg/manifest/workload.go

```diff
@@ -0,0 +1,273 @@
+	}
+
+	if err := unmarshal(&b.BuildString); err != nil {
+		return errUnmarshalBuildOpts
+	}
+	return nil
+}
+
+// DockerBuildArgs represents the options specifiable under the "build" field
+// of Docker Compose services. For more information, see:
+// https://docs.docker.com/compose/compose-file/#build
+type DockerBuildArgs struct {
+	Context    *string           `yaml:"context,omitempty"`
+	Dockerfile *string           `yaml:"dockerfile,omitempty"`
+	Args       map[string]string `yaml:"args,omitempty"`
+}
+
+func (b *DockerBuildArgs) isEmpty() bool {
+	if b.Context == nil && b.Dockerfile == nil && b.Args == nil {
+		return true
+	}
+	return false
+}
+
+// LogConfig holds configuration for Firelens to route your logs.
+type LogConfig struct {
+	Image          *string           `yaml:"image"`
+	Destination    map[string]string `yaml:"destination,flow"`
+	EnableMetadata *bool             `yaml:"enableMetadata"`
+	SecretOptions  map[string]string `yaml:"secretOptions"`
+	ConfigFile     *string           `yaml:"configFilePath"`
+}
+
+func (lc *LogConfig) logConfigOpts() *template.LogConfigOpts {
+	return &template.LogConfigOpts{
+		Image:          lc.image(),
+		ConfigFile:     lc.ConfigFile,
+		EnableMetadata: lc.enableMetadata(),
+		Destination:    lc.Destination,
+		SecretOptions:  lc.SecretOptions,
+	}
+}
+
+func (lc *LogConfig) image() *string {
+	if lc.Image == nil {
+		return aws.String(defaultFluentbitImage)
+	}
+	return lc.Image
+}
+
+func (lc *LogConfig) enableMetadata() *string {
+	if lc.EnableMetadata == nil {
+		// Enable ecs log metadata by default.
+		return aws.String("true")
+	}
+	return aws.String(strconv.FormatBool(*lc.EnableMetadata))
+}
+
+// Sidecar holds configuration for all sidecar containers in a service.
+type Sidecar struct {
+	Sidecars map[string]*SidecarConfig `yaml:"sidecars"`
+}
+
+// Options converts the service's sidecar configuration into a format parsable by the templates pkg.
+func (s *Sidecar) Options() ([]*template.SidecarOpts, error) {
+	if s.Sidecars == nil {
+		return nil, nil
+	}
+	var sidecars []*template.SidecarOpts
+	for name, config := range s.Sidecars {
+		port, protocol, err := parsePortMapping(config.Port)
+		if err != nil {
+			return nil, err
+		}
+		sidecars = append(sidecars, &template.SidecarOpts{
+			Name:       aws.String(name),
+			Image:      config.Image,
+			Port:       port,
+			Protocol:   protocol,
+			CredsParam: config.CredsParam,
+		})
+	}
+	return sidecars, nil
+}
+
+// SidecarConfig represents the configurable options for setting up a sidecar container.
+type SidecarConfig struct {
+	Port       *string `yaml:"port"`
+	Image      *string `yaml:"image"`
+	CredsParam *string `yaml:"credentialsParameter"`
+}
+
+// TaskConfig represents the resource boundaries and environment variables for the containers in the task.
+type TaskConfig struct {
+	CPU       *int              `yaml:"cpu"`
+	Memory    *int              `yaml:"memory"`
+	Count     Count             `yaml:"count"`
```

Comment:
I left this in because there's no particular reason that scheduled jobs can't have a count, and pushing this into a separate inlined struct requires us to adjust about 40 separate unit tests. I think it's okay for TaskConfig generically to have a count field, and even for the scheduled job manifest to silently support it, since the `task run` api also allows custom counts and there's no reason to curtail it.